### PR TITLE
Adding data persistence to POST  /api/tests

### PR DESCRIPTION
**Pull request by tdronkers04**
db Query functions `addNewTest` and `addNewTestAssertions` save test configuration details to the the `tests` and `assertions` tables within RDS,.

route accepts json body in the following shape:

```
{
  "test": {
    "title": "tim-full-2",
    "locations": [
      "us-east-1"
    ],
    "minutesBetweenRuns": 200,
    "type": "API",
    "http_request": {
      "method": "GET",
      "url": "https://trellific.corkboard.dev/api/boards/62aa25c9899c9ef1522baaff",
      "assertions": {
        "statusCode": {
          "comparison": "equals",
          "target": 200
        },
        "responseTime": {
          "comparison": "less_than",
          "target": 925
        },
        "headers": [
            {
                "property": "Content-Type",
                "comparison": "equals",
                "target": "application json"
            },
            {
                "property": "Connection",
                "comparison": "equals",
                "target": "keep-alive"
            }
        ],
        "jsonBody": [
            {
                "property": "title",
                "comparison": "equals",
                "target": "Test board #2"
            },
            {
                "property": "lists",
                "comparison": "is_not_empty"
            }
        ]
      }
    }
  }
}
```

View of results from Sqlectron:

## tests
```
[
  {
    "id": 65,
    "name": "tim-full-2-us-east-1",
    "run_frequency_mins": 200,
    "method": "GET",
    "url": "https://trellific.corkboard.dev/api/boards/62aa25c9899c9ef1522baaff",
    "headers": null,
    "payload": null,
    "query_params": null,
    "teardown": null,
    "status": "enabled",
    "created_at": "2022-07-12 21:00:53.09164",
    "updated_at": null
  }
]
```

## assertions
```
[
  {
    "id": 152,
    "test_id": 65,
    "type": "statusCode",
    "property": "null",
    "comparison_type_id": 4,
    "expected_value": "200"
  },
  {
    "id": 153,
    "test_id": 65,
    "type": "responseTime",
    "property": "null",
    "comparison_type_id": 1,
    "expected_value": "925"
  },
  {
    "id": 154,
    "test_id": 65,
    "type": "headers",
    "property": "Content-Type",
    "comparison_type_id": 4,
    "expected_value": "application json"
  },
  {
    "id": 155,
    "test_id": 65,
    "type": "headers",
    "property": "Connection",
    "comparison_type_id": 4,
    "expected_value": "keep-alive"
  },
  {
    "id": 156,
    "test_id": 65,
    "type": "jsonBody",
    "property": "title",
    "comparison_type_id": 4,
    "expected_value": "Test board #2"
  },
  {
    "id": 157,
    "test_id": 65,
    "type": "jsonBody",
    "property": "lists",
    "comparison_type_id": 7,
    "expected_value": "null"
  }
]
```

### CODE REVIEW

**Comment by tdronkers04** [comment]
Open To-Dos:

refactor testsController regarding creation of a test for each region (current functionality); based on new architecture only one rule is needed for multiple regions

File: controllers/testsController.js [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
       ruleName,
       lambdaArn: LOCATION_TO_TEST_RUNNER[location].arn,
       lambdaName: LOCATION_TO_TEST_RUNNER[location].title,
       inputJSON: JSON.stringify(test.http_request),
     });
 
-    return targetResponse;
+    const result = await addNewTest(ruleName, test);
+
+    if (!result) {
```
Comment (scogra17) [inline controllers/testsController.js]: Wondering if it would be more proper to wrap the `addNewTest` in a try-catch?

File: lib/db/query.js [hunk 2]
```diff
@@ -1,14 +1,57 @@
...
+  let queryCopy = query.slice(0, query.length - 1);
+  queryCopy += ';';
+  const result = await dbQuery(queryCopy);
+
+  if (result.rowCount === 0) return false;
+  return true;
+}
+
+async function addNewTest(ruleName, test) {
+  const { minutesBetweenRuns, http_request: httpRequest } = test;
```
Comment (scogra17) [inline lib/db/query.js]: I think we ultimately want to migrate the input to be "httpRequest" (check out the event.json [here](https://github.com/No-Name-temporary/test-runner-prototype/pull/3)). If we leave as is, can you add in a commented TODO?

**Review by mabbason — COMMENTED** [review]
Looks good Tim! I will leave the explicit approval for someone else. Is the commented code for your next PR then prolly? Good work 👍

File: lib/db/query.js [hunk 3]
```diff
@@ -1,14 +1,57 @@
...
+  let query = `
+    INSERT INTO assertions (test_id, type, property, comparison_type_id, expected_value)
+    VALUES
+  `;
+  Object.keys(assertions).forEach((key) => {
+    if (Array.isArray(assertions[key])) { // headers && jsonBody
+      const detailsArr = assertions[key];
+      detailsArr.forEach((subAssertion) => {
+        const comparisonTypeId = comparisonTypeIds[subAssertion.comparison];
+        query += `(${testId}, '${key}', '${subAssertion.property || null}', ${comparisonTypeId}, '${subAssertion.target || null}'),`;
```
Comment (scogra17) [inline lib/db/query.js:22]: Thoughts on adding a commented TODO to fix the quoted null?

File: lib/db/query.js [hunk 4]
```diff
@@ -1,14 +1,57 @@
...
+      });
+    } else { // responseTime and statusCode
+      const details = assertions[key];
+      const comparisonTypeId = comparisonTypeIds[details.comparison];
+      query += `(${testId}, '${key}', '${details.property || null}', ${comparisonTypeId}, '${details.target || null}'),`;
```
Comment (scogra17) [inline lib/db/query.js:28]: Same as above ^

**Comment by scogra17** [comment]
Overall looks awesome 👍 Nice work! I added a few comments but feel free to disagree and ignore :)

**Review by scogra17 — APPROVED** [review]

File: controllers/testsController.js (hunk 1 above)
Comment (tdronkers04) [reply controllers/testsController.js]: yes, agreed. good call

File: lib/db/query.js (hunk 2 above)
Comment (tdronkers04) [reply lib/db/query.js]: done. also modified the expected payload to read `httpRequest` in the docs

**Pull request merged**